### DISCUSSION TITLE
Using conditional middleware

### DISCUSSION THREAD

**Discussion by mohsenbostan** [spatie/laravel-permission#1818] [Q&A] [answered]
Hi, I am trying to use conditional middleware.
I have some permissions that the package will handle. and in my users table, I have a field called *is_creator*.
I have a route that should only be accessible by people who have the relevant permission or when the *is_creator* field in the database is set to true.

Would you please tell me how to handle this properly?

**Comment by erikn69** [comment, marked as answer]
maybe https://spatie.be/docs/laravel-permission/v4/basic-usage/super-admin#gatebefore

So you can check
```php
return $user->is_creator && request()->is('your_route');
```

**Reply by mohsenbostan** [reply]
It seems like a dirty way to do things, doesn't it?

**Reply by erikn69** [reply]
Yea, but it is in the docs, and make the work, but other ways to do that are more "dirty"
You can wait for another solution

**Reply by mohsenbostan** [reply]
thanks 🙏

**Reply by drbyte** [reply]
Your application needs to enforce the required rules.
This package only provides responses in relation to roles/permissions that it tracks.

So, your application code can combine your own application logic (ie: test for `is_creator` property) and role/permission from this package (`hasRole('foo')`), and any other logic your application needs.
You are welcome to put that into a custom middleware if you wish.

**Reply by mohsenbostan** [reply]
I don't have any other options, thanks.
